packages/components/src/List/List.component.js

```diff
@@ -32,7 +33,14 @@ function ListToolbar({ id, toolbar, displayMode, list, renderers }) {
 		};
 	}
 
-	return <Toolbar {...toolbarProps} sort={!shouldHideSortOptions && toolbarProps.sort} />;
+	return (
+		<Toolbar
+			{...toolbarProps}
+			sort={!shouldHideSortOptions && toolbarProps.sort}
+			getComponent={getComponent}
```

Comment:
You spread toolbarProps with getComponent and you pass again getComponent & components. You could remove them.